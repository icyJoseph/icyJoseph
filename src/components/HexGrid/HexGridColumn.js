import React from "react";
import PropTypes from "prop-types";
import { Grid } from "semantic-ui-react";
import Hexagon from "react-hexagon";
import { curry } from "../../functional";

export const HexGridColumn = ({ tile, clickHandler }) => {
  const { id, Component, fill, backgroundFill } = tile;
  const injectHandlerWithContent = curry(clickHandler)(id);
  return (
    <Grid.Column>
      <Hexagon
        style={{ stroke: "none", fill: backgroundFill }}
        diagonal={100}
        onClick={injectHandlerWithContent}
        children={<Component handler={injectHandlerWithContent} fill={fill} />}
      />
    </Grid.Column>
  );
};

export default HexGridColumn;

HexGridColumn.propTypes = {
  tile: PropTypes.shape({
    Component: PropTypes.func,
    fill: PropTypes.string,
    backgroundFill: PropTypes.string,
    Content: PropTypes.func
  }),
  clickHandler: PropTypes.func
};
