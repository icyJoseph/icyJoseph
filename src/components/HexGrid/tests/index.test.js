import React from "react";
import { shallow } from "enzyme";
import Hexagon from "react-hexagon";

import HexGridColumn from "../HexGridColumn";

describe("HexGridColumn", () => {
  const spy = jest.fn();
  const props = {
    tile: {
      Component: () => <div>Hi</div>,
      Content: <div>hi</div>,
      fill: "#222",
      backgroundFill: "#333"
    },
    clickHandler: spy
  };

  const wrapper = shallow(<HexGridColumn {...props} />);

  it("renders", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("onClick, Hexagon uses clickHandler", () => {
    wrapper.find(Hexagon).simulate("click");
    expect(spy).toHaveBeenCalledWith(props.tile.Content);
  });

  it("injects clickHanlder with Content into Hexagon", () => {
    const onClick = wrapper.find(Hexagon).prop(["onClick"]);
    expect(onClick).toBeInstanceOf(Function);
    onClick();
    expect(spy).toHaveBeenCalledTimes(2);
  });
});
