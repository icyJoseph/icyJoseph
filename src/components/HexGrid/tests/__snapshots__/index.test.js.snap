// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HexGridColumn renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HexGridColumn
    clickHandler={[Function]}
    tile={
        Object {
            "Component": [Function],
            "Content": <div>
              hi
        </div>,
            "backgroundFill": "#333",
            "fill": "#222",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <Hexagon
        diagonal={100}
        flatTop={false}
        onClick={[Function]}
        style={
                Object {
                        "fill": "#333",
                        "stroke": "none",
                      }
        }
>
        <Component
                fill="#222"
                handler={[Function]}
        />
</Hexagon>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Component
          fill="#222"
          handler={[Function]}
/>,
        "diagonal": 100,
        "flatTop": false,
        "onClick": [Function],
        "style": Object {
          "fill": "#333",
          "stroke": "none",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "fill": "#222",
          "handler": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Hexagon
          diagonal={100}
          flatTop={false}
          onClick={[Function]}
          style={
                    Object {
                              "fill": "#333",
                              "stroke": "none",
                            }
          }
>
          <Component
                    fill="#222"
                    handler={[Function]}
          />
</Hexagon>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Component
            fill="#222"
            handler={[Function]}
/>,
          "diagonal": 100,
          "flatTop": false,
          "onClick": [Function],
          "style": Object {
            "fill": "#333",
            "stroke": "none",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "fill": "#222",
            "handler": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
